app/controllers/products_controller.rb

```diff
@@ -0,0 +1,57 @@
+class ProductsController < ApplicationController
+
+  def index
+    if params[:price].present? || params[:quantity].present?
+      @products = Product.where('price = :price OR total_quantity = :quantity', price: params[:price].to_f,
+      quantity: params[:quantity].to_i)
+
+    elsif params[:price].present? && params[:quantity].present?
+      @products = Product.where('price = :price AND total_quantity = :quantity', price: params[:price].to_f,
+      quantity: params[:quantity].to_i)
+
```

Comment:
Remove this extra line

Reply:
Let's add @products = Product.all in the start. 
Then, add params[:price].present? && params[:quantity].present? as first if condition

Right now, it will never come to elsif condition
Also, as these are queries, so they should be in model, not the controller.

Reply:
Let's discuss this to avoid confusions